create-docker-action.py

```diff
@@ -0,0 +1,78 @@
+import json
+import os
+import pathlib
+
+DESCRIPTION = 'description'
+REQUIRED = 'required'
+
+EVENT = os.environ['EVENT']
+REF = os.environ['REF']
+REPO = os.environ['REPO']
+REPO_OWNER_ID = os.environ['REPO_OWNER_ID']
+REPO_OWNER_ID_PYPA = '647025'
+
+
+def set_image(event: str, ref: str, repo: str, repo_owner_id: str) -> str:
+    if event == 'pull_request' and repo_owner_id == REPO_OWNER_ID_PYPA:
+        return '../../../Dockerfile'
+    docker_ref = ref.replace('/', '-')
+    return f'docker://ghcr.io/{repo}:{docker_ref}'
+
+
+image = set_image(EVENT, REF, REPO, REPO_OWNER_ID)
+
+action = {
+    'name': '🏃',
+    DESCRIPTION: (
+        'Run Docker container to upload Python distribution packages to PyPI'
+    ),
+    'inputs': {
```

Comment:
Since Python 3.11, there's `tomllib` in stdlib. Let's use it to read the contents of `action.yml` and paste the `inputs` parsed from it here.

Reply:
> Since Python 3.11, there's `tomllib` in stdlib. Let's use it to read the contents of `action.yml` and paste the `inputs` parsed from it here.

I don't think that will work because the file is not formatted as TOML. Attempting to load the file raises `tomllib.TOMLDecodeError`.

```text
gh-action-pypi-publish on ghcr via 🐍 v3.12.3
❯ python create-docker-action.py
Traceback (most recent call last):
  File "/Users/brendon/dev/gh-action-pypi-publish/create-docker-action.py", line 25, in <module>
    composite_action_contents = tomllib.loads(composite_action_text)
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/tomllib/_parser.py", line 102, in loads
    pos = key_value_rule(src, pos, out, header, parse_float)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/tomllib/_parser.py", line 326, in key_value_rule
    pos, key, value = parse_key_value_pair(src, pos, parse_float)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/tomllib/_parser.py", line 366, in parse_key_value_pair
    raise suffixed_err(src, pos, "Expected '=' after a key in a key/value pair")
tomllib.TOMLDecodeError: Expected '=' after a key in a key/value pair (at line 1, column 4)
```

Reply:
Right. I should stop reviewing in the middle of the night :)
Marking as resovled.